docs/rfcs/016-asset-metadata.md

```diff
@@ -0,0 +1,95 @@
+
+This table will initially store the available thumbnail sizes, replicating/duplicating what is stored in `s.json`, but can easily be read as part of a database query. These can be read in bulk alongside the corresponding `Asset` record.
+
+The handling of a NamedQuery is fairly complex to allow for query building reuse. Currently reading metadata is only required for manifest projection so we will need to add a hook in the processing to add the required `.Include()` where appropriate.
+
+### Future Improvements
+
+While we are only storing thumbnailSizes now, this new table could be used to store a variety of values in the future. Some examples are:
+
+* Generated transcode types and output locations for AV.
+* For `file` delivery channel - do we store a copy of the original file? If so, where.
+* For images - do we store a copy of the file? Is it original (`use-original`) or a transcode?
+* Adjuncts: what is stored where?
+* Checksum of Asset origin - could help to identify when source image has been updated.
+* Periodic request metrics. An external request could calculate metrics and periodically write summary back to db (per day/month/year). 
+
+The above values can then be used to drive generation of improved [single-item manifest](https://github.com/dlcs/protagonist/issues/488) and clearing up [no-longer required delivery artifacts](https://github.com/dlcs/protagonist/issues/430).
+
+### Database
+
+#### Table Name
+
+Considered names and reason for choosing or not:
+* `AssetApplicationMetadata` - chosen name as doesn't add restriction to what is being stored but `Application` in name highlights that this is for internal metadata only, not replacements for string1, string2 etc.
+* `ImageMetadata` - while suitable this is vague and opens up the table to be a dumping ground for any data.
+* `DeliveryChannelMetadata` - originally considered name, this would point to having both an `assetId` and `channel` record per row but some values may be relevant for multiple channels (namely image + thumbs). We may also want to store some data that is not delivery-channel specified (e.g. checksum).
+  * If we opted for this table we could consider adding a new column to the [`ImageDeliveryChannel`](014-delivery-channels-database.md) table, as this would store a row per Asset/Delivery channel.
+* `AssetDeliveryMetadata` - similar to above - if we are storing checksum etc this isn't asset delivery (ie Orchestrator/Thumbs) specific.
+
+#### Schema
+
+The suggested schema for the table should be flexible.
+
+| Column        | Type        | Description                         |
+| ------------- | ----------- | ----------------------------------- |
+| AssetId       | text        | AssetId this is for                 |
+| MetadataType  | text        | Identifier for the type of metadata |
+| MetadataValue | jsonb       | JSON object of values for type      |
+| {audit-cols}  | timestamptz | Created/updated dates               |
+

```

Comment:
(for later) This table might store information about the use of assets in IIIF Manifests.